data_types/src/database_rules.rs

```diff
@@ -199,6 +199,9 @@ pub struct LifecycleRules {
     /// Allow dropping data that has not been persisted to object storage
     pub drop_non_persisted: bool,
 
+    /// Allow chunks to be persisted to object storage.
```

Comment:
Does this setting force / ensure that chunks are persisted to object storage before drop? If so, perhaps we should remove the `Allow` which makes it sound to me as if if persisting chunks is optional.

```suggestion
    /// Persists chunks to object storage.
```